dcc-storage-server/src/main/java/org/icgc/dcc/storage/server/controller/ObjectUploadController.java

```diff
@@ -149,23 +161,55 @@ public void tryRecover(
     return uploadService.getUploadStatus(objectId, uploadService.getUploadId(objectId), fileSize);
   }
 
+  @ProjectCodeScoped
   @RequestMapping(method = RequestMethod.GET, value = "/{object-id}")
   public @ResponseBody Boolean isObjectExist(@RequestHeader(HttpHeaders.AUTHORIZATION) final String accessToken,
       @PathVariable("object-id") String objectId) {
     return uploadService.exists(objectId);
   }
 
+  @ProjectCodeScoped
   @RequestMapping(method = RequestMethod.DELETE, value = "/{object-id}")
   @ResponseStatus(value = HttpStatus.OK)
   public void cancelUpload(@RequestHeader(HttpHeaders.AUTHORIZATION) final String accessToken,
       @PathVariable("object-id") String objectId) {
     uploadService.cancelUpload(objectId, uploadService.getUploadId(objectId));
   }
 
+  @ProjectCodeScoped
   @RequestMapping(method = RequestMethod.POST, value = "/cancel")
   @ResponseStatus(value = HttpStatus.OK)
   public void cancelAll()
       throws IOException {
     uploadService.cancelUploads();
   }
+
+  @ProjectCodeScoped
+  @RequestMapping(method = RequestMethod.GET, value = "/test/{object-id}")
+  @ResponseStatus(value = HttpStatus.OK)
+  public @ResponseBody String test(
+      @RequestHeader(value = HttpHeaders.AUTHORIZATION) final String accessToken,
+      @PathVariable(value = "object-id") String objectId) {
+    log.info("Test invoked!");
+    return "Upload Test Operation executed";
+  }
+
+  /**
+   * Exception handler specific to the Spring Security processing in this controller
+   * @return Error if Spring Security policies are violated
+   */
+  @ExceptionHandler({ AccessDeniedException.class })
+  public ResponseEntity<Object> handleAccessDeniedException(HttpServletRequest req, AccessDeniedException ex) {
```

Comment:
Why are we repeating the same handler in 2 different classes. Better to use `@ControllerAdvice`